HaxmDriver.sln

```diff
@@ -15,38 +15,42 @@ Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "IntelHaxm", "windows\IntelH
 EndProject
 Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "haxlib", "core\haxlib.vcxproj", "{BC80D1E0-5738-4048-A742-8A20949A6587}"
 EndProject
+Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Tests", "Tests", "{4FF4BA9A-1A63-4CBF-A6C0-1D2D83B2146E}"
+EndProject
+Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "tests", "tests\tests.vcxproj", "{0458723C-5C3F-4509-BDAE-35FAE2A90C19}"

```

Comment:
There doesn't seem to be an easy way to exclude `tests` from the VS solution build yet. I wonder how we can add a "no tests" option to the build. It's okay if `tests` is built by default, but it's not really a strict requirement for the driver build, so we should make it optional.

Actually I'm having a lot of trouble installing all the dependencies for `tests`... Still haven't got `keystone` installed via `vcpkg` due to a CMake failure :(

Reply:
That sounds great! Hopefully NuGet maintainers are more responsive. I like the idea of vcpkg, but right now I don't think it's mature enough for our project to depend on...

Reply:
Sorry for dropping that PR on the floor! With our recent support for Linux and OSX, we built up a backlog of PRs that we're actively trying to clear.

Vcpkg can also be used to produce NuGet packages containing all your dependencies automatically: `vcpkg export --nuget keystone gtest`

Reply:
@ras0219-msft No worries, thanks for the `export --nuget` hint.

Include paths are not working with the exported package. This are the steps to repeat it:
1. `vcpkg export keystone:x86-windows keystone:x64-windows --nuget --nuget-id=KeystoneEngine --nuget-version=0.9.1`
2. Update the *.nupkg file just to add a description, icon, etc. to the *.nuspec file.
3, Create a new VS solution+project, and install the *.nupkg file from a local repository.
4. Add `#include <keystone/keystone.h>` as press build.

Somehow MSVC complains that it didn't find that file, on all {Debug,Release} {Win32,x86,x64} modes, which is strange considering that the *.vcxproj file automatically added the `packages\keystoneengine.0.9.1\build\native\keystoneengine.targets` file.

Reply:
I noticed something last week, although I'm not sure if it's related to this problem: `HaxmDriver.sln` uses build configurations like `Win7 Debug`, `Win7 Release`, etc., but not `Debug`/`Release`. However, vcpkg seems to look for the latter:

```
> msbuild HaxmDriver.sln /p:Configuration="Win7 Debug" /p:Platform="x64"
...
VcpkgTripletSelection:
  Using triplet "x64-windows" from "<redacted>\vcpkg\installed\x64-windows\"
  Vcpkg is unable to link because we cannot decide between Release and Debug libraries. Please define the property VcpkgConfiguration to be 'Release' or 'Debug' (currently 'Win7 Debug').
...
```

Although these warnings do not fail the build.

This PR has actually added `Debug` and `Release` to `HaxmDriver.sln`, but I don't think they are usable now. Probably the right thing to do is map them to `Win7 Debug` and `Win7 Release`, respectively. Then I guess our VS solution will work better with vcpkg.